ada_feeding/ada_feeding/behaviors/acquisition/compute_food_frame.py

```diff
@@ -5,56 +5,246 @@
+        debug_food_frame (string): If len>0, TF frame to publish static transform
+                                   (relative to world_frame) for debugging purposes
+        """
+        # pylint: disable=unused-argument
+        # Arguments are handled generically in base class.
+        super().blackboard_inputs(
+            **{key: value for key, value in locals().items() if key != "self"}
+        )
 
-class FlipFoodFrame(py_trees.behavior.Behavior):
-    """
-    TODO
-    Take the food frame from the blackboard and 
-    rotate it PI about the world frame +Z axis.
-    """
-
-    def __init__(
+    def blackboard_outputs(
         self,
-        name: str,
-        node: Node,
-        frame_blackboard_key: str,
-        world_frame_id: str,
+        action_select_request: Optional[BlackboardKey],  # AcquisitionSelect.Request
+        food_frame: Optional[BlackboardKey],  # TransformStamped
+        debug_tf_publisher: Optional[
+            BlackboardKey
+        ] = None,  # StaticTransformBroadcaster
     ) -> None:
-        # Initiatilize the behavior
-        super().__init__(name=name)
-        pass
+        """
+        Blackboard Outputs
+        By convention (to avoid collisions), avoid non-None default arguments.
+
+        Parameters
+        ----------
+        action_select_request (AcquisitionSelect.Request): request to send to AcquisitionSelect
+                                                           (copies mask input)
+        food_frame (geometry_msgs/TransformStamped): transform from world_frame to food frame
+        debug_tf_publisher (StaticTransformBroadcaster): If set, store
+                                                        static broadcaster here to keep it alive
+                                                        for debugging purposes.
+        """
+        # pylint: disable=unused-argument
+        # Arguments are handled generically in base class.
+        super().blackboard_outputs(
+            **{key: value for key, value in locals().items() if key != "self"}
+        )
+
+    # pylint: disable=attribute-defined-outside-init
+    # It is okay for attributes in behaviors to be
+    # defined in the setup / initialise functions.
+
+    def setup(self, **kwargs):
+        """
+        Middleware (i.e. TF) setup
+        """
+
+        # Create TF Tree
+        self.tf_buffer = Buffer()
+        self.tf_listener = TransformListener(
+            self.tf_buffer, self.blackboard_get("ros2_node")
+        )
 
-    def setup(self, **kwargs) -> None:
-        pass
+    def initialise(self):
+        """
+        Behavior initialization
+        """
+
+        # Construct camera intrinsics
+        # See: https://medium.com/@yasuhirachiba/converting-2d-image-coordinates-
+        #      to-3d-coordinates-using-ros-intel-realsense-d435-kinect-88621e8e733a
+        camera_info = self.blackboard_get("camera_info")
+        self.intrinsics = pyrealsense2.intrinsics()
```

Comment:
If this library works well for you, we should add an Issue to use it in FaceDetection as well. Currently, FaceDetection implements its own intrinsics calibration.